src/Collection.ts

```diff
@@ -77,27 +79,26 @@ class Collection extends Resource {
 
     // eager loading of 'fetchAllUri' (e.g. parent for embedded collections)
     if (this.config.avoidNPlusOneRequests) {
-      return LoadingCollection.create(itemsLoaded)
+      return LoadingCollection.create<ItemType>(itemsLoaded)
 
       // no eager loading: replace each reference (Link) with a Resource (ResourceInterface)
     } else {
-      return LoadingCollection.create(itemsLoaded, this._replaceEntityReferences(array))
+      return LoadingCollection.create<ItemType>(itemsLoaded, this._replaceEntityReferences(array))
     }
   }
 
   /**
    * Replace each item in array with a proper Resource (or LoadingResource)
    */
-  private _replaceEntityReferences (array: Array<Link>): Array<ResourceInterface> {
-    return array
-      .filter(entry => isEntityReference(entry))
-      .map(entry => this.apiActions.get(entry.href))
+  private _replaceEntityReferences (array: Array<ItemType>): Array<ItemType> {
```

Comment:
Is not from your change, but this method is dangerous: if filters out the items which are not Link, and removes them silently from the collection

Reply:
Can you create an Issue to maybe warn the user?